common4j/src/main/com/microsoft/identity/common/java/ui/PreferredAuthMethod.kt

```diff
@@ -23,13 +23,24 @@
 package com.microsoft.identity.common.java.ui
 
 /**
- * Preferred authentication method for the user.
+ *  Authentication methods for the user.
  * This code will be sent to eSTS as a hint to what authentication method the user prefers.
  * If not specified, eSTS will use the default authentication method.
  */
-enum class PreferredAuthMethod(@JvmField val code: Int) {
+enum class PreferredAuthMethod(@JvmField val code: Int, @JvmField val value: String?) {
+    /**
+     * No preferred authentication method.
+     */
+    NONE(0, null),
+
     /**
      * QR code + PIN authentication.
      */
-    QR(18)
+    QR(18, "qrpin");
+    companion object {
+        @JvmStatic
+        fun fromCode(code: Int) = PreferredAuthMethod.values().first { it.code == code }
```

Comment:
If a new one is introduced in the future, 
(let's say SOME_NEW_METHOD(30,"newmethod"))

the change is pushed to broker, but the MSAL side is still not updated.

Casting here would fail given that code "30" doesn't exist. How would we gracefully handle such scenario?

Reply:
I'm wondering if it would make sense to return a prioritized list instead of just one value, and the SDK side can pick the highest priority one that it supports.

(disclaimer: i don't have insight knowledge on how "PreferredAuthMethod" really works)

Reply:
I think it makes sense to return (1) a list or at least and (2) exception indicating the preferred auth method is not supported on this version and that an update is required.

For point 1 I will need to discuss this with IOS and one Auth 
Point 2 nevertheless I can add it.

Reply:
for 1 we decide to keep it as it is for this PP
for 2 I ipdate code to send an exception indicating an update is required.